bitcoin/src/network/constants.rs

```diff
@@ -119,6 +119,28 @@ impl Network {
             Network::Regtest => "regtest",
         }
     }
+
+    /// Converts a `bitcoind -chain` argument name to its equivalent `Network`.
+    ///
+    /// ```bash,no_run
+    /// $ bitcoin-23.0/bin/bitcoind --help | grep -C 3 '\-chain=<chain>'
+    /// Chain selection options:
+    ///
+    /// -chain=<chain>
+    /// Use the chain <chain> (default: main). Allowed values: main, test, signet, regtest
+    /// ```
+    pub fn from_core_arg(core_arg: &str) -> Result<Self, ParseNetworkError> {
+        use Network::*;
+
+        let network = match core_arg {
+           "main" | "mainnet" => Bitcoin,
+           "test" | "testnet" => Testnet,
```

Comment:
Were "mainnet" and "testnet" used in old versions of bitcoind? 22 and 23 use "main" and "test", I don't have any older versions laying around.

Reply:
AFAIR `mainnet` was never used, not sure about `testnet`. I'd rather not accept strings that aren't used in Core.

Also does anyone remember if Core is case-sensitive? My RAM is full RN so can't spawn a VM.

Reply:
I don't think they were ever used, we may not want to have this in `from_core_arg` but we may for `from_str`.

Reply:
There are two distinct ways of using bitcoind/bitcoin-cli. As Sr-gi mentions, 

1) As an named option `chain=` `bitcoind -chain={main, test, regtest, signet}`
2) Directly as an unnamed option `bitcoind --testnet` / `bitcoind --signet`, / `bitcoind --regtest`

Given this, I agree with the current implementation of this PR.

Reply:
For cross refernce: https://github.com/bitcoin/bitcoin/blob/cac29f5cd66f005ce65edd697990d495a904a1f1/src/chainparamsbase.cpp#L18-L29

Reply:
I don't think that is a good reason since the documentation explicitly mentions `-chain` argument in both `to_core_arg` and `from_core_arg`. I don't really like accepting `-` variants since they can't represent mainnet and we would have to require `-`. For something exotic such as Rust replacement of `bitcoin-cli` people should just write their own I think.

Perhaps we should rename the functions to `{to,from}_core_chain_arg` to be entirely clear but could be annoying.